cron/sqs_status.py

```diff
@@ -0,0 +1,116 @@
+            eventTaskQueue = mqConn.get_queue(exchange)
+            # get queue count
+            taskQueueAttributes = eventTaskQueue.count()
+            taskQueueNotVisible = eventTaskQueue.get_attributes('ApproximateNumberOfMessagesNotVisible')
+            # Build up our dict
+            if exchange not in sqslist:
+                sqslist[exchange] = {}
+                sqslist[exchange]['messages'] = taskQueueAttributes
+                sqslist[exchange]['inflight'] = int(taskQueueNotVisible['ApproximateNumberOfMessagesNotVisible'])
+
+
+        # setup a log entry for health/status.
+        healthlog = dict(
+            utctimestamp=toUTC(datetime.now()).isoformat(),
+            hostname='server',
+            processid=os.getpid(),
+            processname=sys.argv[0],
+            severity='INFO',
+            summary='mozdef health/status',
+            category='mozdef',
+            source='aws-sqs',
+            tags=[],
+            details=[])
+
+        healthlog['details'] = dict(username='mozdef')
+        healthlog['details']['queues']= sqslist
+        healthlog['details']['total_messages_ready'] = 0
+        healthlog['details']['total_feeds'] = 0
+        healthlog['tags'] = ['mozdef', 'status', 'sqs']
+        feeds = len(sqslist.keys())
+        healthlog['details']['total_feeds'] = feeds
+        ready = 0
+        for exchange in options.taskexchange:
+            ready1 = sqslist[exchange]['messages']
+            ready = ready1 + ready
+        healthlog['details']['total_messages_ready'] = ready
+        es.save_event(index=options.index, doc_type='mozdefhealth', body=json.dumps(healthlog))
+    except Exception as e:
+        logger.error("Exception %r when gathering health and status " % e)
+
+def main():
+    logger.debug('Starting')
+    logger.debug(options)
+    getQueueSizes()
+
+def initConfig():
+    # aws options
+    options.accesskey = getConfig('accesskey', '', options.configfile)
+    options.secretkey = getConfig('secretkey', '', options.configfile)
+    options.region = getConfig('region', 'us-west-1', options.configfile)
+    options.taskexchange = getConfig('taskexchange', 'nsmglobalsqs', options.configfile).split(',')
+    options.output = getConfig('output', 'stdout', options.configfile)
+    options.sysloghostname = getConfig('sysloghostname', 'localhost', options.configfile)
+    options.syslogport = getConfig('syslogport', 514, options.configfile)
+    options.esservers = list(getConfig('esservers', 'http://localhost:9200', options.configfile).split(','))
+    options.index = getConfig('index', 'mozdefstate', options.configfile)
+
+if __name__ == '__main__':
+    # configure ourselves
+    parser = OptionParser()
+    parser.add_option("-c", dest='configfile', default=sys.argv[0].replace('.py', '.conf'), help="configuration file to use")
+    (options, args) = parser.parse_args()
+    initConfig()
+    initLogger(options)
+    getQueueSizes()
```

Comment:
shouldn't this call main() instead of doing what main() is doing?

Reply:
You're absolutely right.